File: 2025adventofcode1.2.py
dialpointsto = 50
past100count = 0
def opensafe(dialchange):
    global dialpointsto
    global past100count
    if dialchange[0] == "L":
        dialchange= int("-" + dialchange[1:])
        for x in range (0, dialchange, -1):
            dialpointsto -= 1
            if dialpointsto % 100 == 0:
                past100count +=1
    else:
        dialchange = int(dialchange[1:])
        for x in range (0, dialchange):
            dialpointsto += 1
            if dialpointsto % 100 == 0:
                past100count +=1
import fileinput
with open('input.txt') as f:
    for line in f:
        dialchange = line.strip()
        opensafe(dialchange)
print(past100count)

File: 2025/c++/2025adventofcode1.2.cpp
// Solution for Advent of Code 2025, Day 1, Part 25.
# include <fstream>
# include <iostream>
# include <string>

int _single_change(std::string dialchange, int& d) {
    char direction = dialchange[0];
    int magnitude = std::stoi(dialchange.substr(1));
    int passes = 0;
    if (direction == 'L') {
        for (int i = 0; i < magnitude; i++) {
            d -= 1;
            if (d % 100 == 0) {
                passes += 1;
            }
        }
    }   else if (direction == 'R') {
        for (int i = 0; i < magnitude; i++) {
            d+=1;
            if (d % 100 == 0) {
                passes += 1;
            }
        }
    }
    return passes;
}

int main() {
    int dial_pos = 50;
    int past_100_count = 0;
    std::fstream input_file("input.txt");
    std::string line;

    while (std::getline(input_file, line)) {
        past_100_count += _single_change(line, dial_pos);
    }

    std::cout << past_100_count << "\n";
    return 0;
}
